cpp/example_code/dynamodb/update_item.cpp

```diff
@@ -1,123 +1,146 @@
-{
-    const std::string USAGE = "\n"
-        "Usage:\n"
-        "    update_item <tableName> <keyValue> <attribute=value> ..\n\n"
-        "Where:\n"
-        "    tableName       - name of the table to put the item in\n"
-        "    keyValue        - the key value to update\n"
-        "    attribute=value - attribute=updated value\n\n"
-        "Examples:\n"
-        "    update_item SiteColors text default=000000\n"
-        "    update_item SiteColors background code=d3d3d3\n\n";
-
-    if (argc < 4)
-    {
-        std::cout << USAGE;
+
+bool AwsDoc::DynamoDB::updateItem(const Aws::String &tableName,
+                                  const Aws::String &partitionKey,
+                                  const Aws::String &partitionValue,
+                                  const Aws::String &attributeKey,
+                                  const Aws::String &attributeValue,
+                                  const Aws::Client::ClientConfiguration &clientConfiguration) {
+    Aws::DynamoDB::DynamoDBClient dynamoClient(clientConfiguration);
+
+    // *** Define UpdateItem request arguments.
+    // Define TableName argument.
+    Aws::DynamoDB::Model::UpdateItemRequest request;
+    request.SetTableName(tableName);
+
+    // Define KeyName argument.
+    Aws::DynamoDB::Model::AttributeValue attribValue;
+    attribValue.SetS(partitionValue);
+    request.AddKey(partitionKey, attribValue);
+
+    // Construct the SET update expression argument.
+    Aws::String update_expression("SET #a = :valueA");
+    request.SetUpdateExpression(update_expression);
+
+    // Construct attribute name argument.
+    Aws::Map<Aws::String, Aws::String> expressionAttributeNames;
+    expressionAttributeNames["#a"] = attributeKey;
+    request.SetExpressionAttributeNames(expressionAttributeNames);
+
+    // Construct attribute value argument.
+    Aws::DynamoDB::Model::AttributeValue attributeUpdatedValue;
+    attributeUpdatedValue.SetS(attributeValue);
+    Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue> expressionAttributeValues;
+    expressionAttributeValues[":valueA"] = attributeUpdatedValue;
+    request.SetExpressionAttributeValues(expressionAttributeValues);
+
+    // Update the item.
+    const Aws::DynamoDB::Model::UpdateItemOutcome &outcome = dynamoClient.UpdateItem(
+            request);
+    if (outcome.IsSuccess()) {
+        std::cout << "Item was updated" << std::endl;
+    }
+    else {
+        std::cerr << outcome.GetError().GetMessage() << std::endl;
+    }
+
+    return outcome.IsSuccess();
+}
+// snippet-end:[dynamodb.cpp.update_item.code]
+
+/*
+ *  main function
+ *
+ *  Usage: 'run_update_item <table_name> <partition_key> <partition_value> <attribute_key>
+ *         <attribute_value>'
+ *
+ *  Prerequisites: A pre-populated DynamoDB table.
+ *
+ *  Instructions for populating a table with sample data can be found at:
+ *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html

```

Comment:
*  For instructions on populating a table with sample data, see
 *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html.